multinet/multinet.py

```diff
@@ -86,26 +87,49 @@ def create_graph(workspace, graph):
     if missing:
         return (missing, "400 Missing Required Parameters")
 
-    # Validate that all referenced tables exist
-    edges = list(db.db(workspace).collection(edge_table).find({}))
-    invalid_from = set(
-        [
-            edge["_from"].split("/")[0]
-            for edge in edges
-            if edge["_from"].split("/")[0] not in node_tables
-        ]
-    )
-    invalid_to = set(
-        [
-            edge["_to"].split("/")[0]
-            for edge in edges
-            if edge["_to"].split("/")[0] not in node_tables
-        ]
-    )
-
-    if invalid_from or invalid_to:
-        error = {"error": "undefined_tables", "detail": list(invalid_from | invalid_to)}
-        return (error, "400 Edge Table Validation Failed")
+    loadedWorkspace = db.db(workspace)
+    if loadedWorkspace.has_graph(graph):
+        return (graph, "409 Graph Already Exists")
```

Comment:
Here I've added an early check for an existing graph of the same name. Since the workspace is loaded anyway, there's no inefficiency here, and can avoid unnecessary work. You can see that the same response is returned below, if the `db.create_graph` function fails. I wasn't sure if it was a good idea to get rid of this check entirely, as I'm not sure if the operation would fail for any other reason, but we are checking the condition already here. There's also the consideration of which error we want the user to see first, that the graph already exists, or that the graph is invalid. I'm interested to hear your input @waxlamp

Reply:
This is good because it's more explicit. We should get rid of the second check below this line, as it's redundant with the check here.

The "graph already exists" error should show up first, since that has to be fixed before anything else can happen. The user workflow would be to receive an error in the client that the graph name already exists, then select a different name, then try again, then possibly receive validation errors, fix those, and finally succeed in creating the graph.

Reply:
Alright, I'll remove that second check.